railties/test/application/rake/dbs_test.rb

```diff
@@ -461,6 +461,178 @@ def db_schema_sql_dump
         ENV["SCHEMA_FORMAT"] = old_env
       end
 
+      test "db:schema:dump loads schema_migrations and dumps them to individual files with sql_filesystem_versions" do
+        add_to_config "config.active_record.schema_format = :sql_filesystem_versions"
+        require "#{app_path}/config/environment"
+
+        Dir.chdir(app_path) do
+          rails "generate", "model", "book", "title:string"
+          rails "db:migrate"
+
+          assert_not File.exist?("db/schema.rb")
+          assert File.exist?("db/structure.sql")
+
+          schema_migrations = Dir["db/schema_migrations/*"].map { |f| File.basename(f) }
+          assert_equal 1, schema_migrations.size
+          assert_match(/\A\d{14}\z/, migration_files.first)
```

Comment:
whops, some leftovers after rename (I rewrote this whole PR like 4 times 😅 )

will resolve shortly